tests/unit/test_partialmerge_app.py

```diff
@@ -0,0 +1,106 @@
+"""Module contains functional requirements for partilmerge app."""
+
+from uuid import uuid4
+
+import pytest
+from pydantic import ValidationError
+
+import corva
+
+ASSET_VALUE_TO_CACHE = str(uuid4())
+RERUN_ASSET_VALUE_TO_CACHE = str(uuid4())
+RAW_EVENT = {
+    "event_type": "partial-well-rerun-merge",
+    "data": {
+        "partial_well_rerun_id": 123,
+        "partition": 95,
+        "rerun_partition": 2,
+        "rerun_mode": "realtime",
+        "start": 1543847760,
+        "end": 1543847760,
+        "asset_id": 1231234,
+        "rerun_asset_id": 2323245,
+        "app_stream_id": 9585,
+        "rerun_app_stream_id": 4745,
+        "version": 1,
+        "app_id": 46,
+        "app_key": "corva.enrichment-wrapper",
+        "app_connection_id": 2457,
+        "rerun_app_connection_id": 2459,
+        "source_type": "drilling",
+        "log_type": "time",
+        "run_until": 1543847760,
+    },
+}
+
+
+def test_merge_event_handler_called_from_stream_app_on_unexpected_event_type_raises_exc(
+    context,
+):
+    """Partial merge app must raise Pydantic validation exception
+    while processing event of unexpected type.
+    """
+
+    @corva.stream
+    def stream_app(event, api, cache):
+        assert False
+
+    @corva.partial_rerun_merge
+    def partialmerge_app(event, api, asset_cache, rerun_asset_cache):
+        return event
+
+    raw_event = dict(RAW_EVENT)
+    raw_event["event_type"] = "unknown_event_type"
+    with pytest.raises(ValidationError) as e:
+        stream_app(raw_event, context)
+    assert "validation error" in str(e.value)
+
+
+def test_merge_event_handler_called_from_stream_app_returns_expected_cache_values(
```

Comment:
Not sure what this test tests:) Mypy ensures that function receives arguments with correct types

Reply:
This test makes sure that cache object user receives actually works (like is able to store data and retrieve it back).